### PR TITLE
Check for bevy_internal imports in CI

### PR DESCRIPTION
**Pull request by ameknite**
# Objective

- Avoid using bevy_internal imports in examples. 

## Solution

- Add CI to check for bevy_internal imports like suggested in https://github.com/bevyengine/bevy/pull/9547#issuecomment-1689377999
- Fix another import

I don't know much about CI so I don't know if this is the better approach, but I think is better than doing a pull request every time I found this lol, any suggestion is welcome.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -333,3 +333,25 @@ jobs:
...
+        run: |
+          errors=""
+          for file in $(find examples -name '*.rs'); do
+              if grep -q "use bevy_internal" "$file"; then
+                  errors+="ERROR: Detected 'use bevy_internal' in $file\n"
+              fi
+          done
+          if [ -n "$errors" ]; then
+              echo "$errors"
+              echo " Avoid importing bevy_internal, it should not be used direclty"
```
Comment (rparrett) [inline .github/workflows/ci.yml:353]:
```suggestion
              echo " Avoid importing bevy_internal, it should not be used directly"
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -333,3 +333,25 @@ jobs:
...
+        shell: bash
+        run: |
+          errors=""
+          for file in $(find examples -name '*.rs'); do
+              if grep -q "use bevy_internal" "$file"; then
+                  errors+="ERROR: Detected 'use bevy_internal' in $file\n"
+              fi
+          done
+          if [ -n "$errors" ]; then
+              echo "$errors"
```
Comment (rparrett) [inline .github/workflows/ci.yml:352]:
```suggestion
              echo -e "$errors"
```

Fixes the line breaks

**Comment by rparrett** [comment]
Cool. Found a couple minor issues and suggested fixes.

internal

**Review by mockersf — APPROVED** [review]

**Comment by mockersf** [comment]
@cart what do you think of adding it to the required checks?

**Review by alice-i-cecile — APPROVED** [review]

**Review by rparrett — APPROVED** [review]

**Review by cart — APPROVED** [review]
Definitely on board for this (and for making it a required check). I've noticed that Rust Analyzer recommends bevy_internal more regularly now for bevy examples. I'll add this to the required checks after merging (required checks are a repo setting)

**Pull request merged**